src/Service.GraphQLBuilder/GraphQLUtils.cs

```diff
@@ -209,24 +212,38 @@ public static ObjectType UnderlyingGraphQLEntityType(IType type)
+        /// <param name="defaultValueFromConfig">String representation of default value defined in runtime config.</param>
+        /// <param name="parameterDefinition">Database schema metadata for stored procedure parameter which include value and value type.</param>
+        /// <returns>Tuple where first item is the string representation of a GraphQLType (e.g. "Byte", "Int", "Decimal")
+        /// and the second item is the GraphQL *type*ValueNode </returns>
+        /// <exception cref="DataApiBuilderException">Raised when parameter casting fails due to unsupported type.</exception>
+        public static Tuple<string, IValueNode> ConvertValueToGraphQLType(string defaultValueFromConfig, ParameterDefinition parameterDefinition)
         {
-            if (int.TryParse(stringValue, out int integerValue))
+            string paramValueType = SchemaConverter.GetGraphQLTypeFromSystemType(type: parameterDefinition.SystemType);
+            Tuple<string, IValueNode> valueNode = paramValueType switch
             {
-                return new(LONG_TYPE, new IntValueNode(integerValue));
-            }
-            else if (double.TryParse(stringValue, out double floatingValue))
-            {
-                return new(FLOAT_TYPE, new FloatValueNode(floatingValue));
-            }
-            else if (bool.TryParse(stringValue, out bool booleanValue))
-            {
-                return new(BOOLEAN_TYPE, new BooleanValueNode(booleanValue));
-            }
+                BYTE_TYPE => new(BYTE_TYPE, new IntValueNode(byte.Parse(defaultValueFromConfig))),
+                SHORT_TYPE => new(SHORT_TYPE, new IntValueNode(short.Parse(defaultValueFromConfig))),
+                INT_TYPE => new(INT_TYPE, new IntValueNode(int.Parse(defaultValueFromConfig))),
+                LONG_TYPE => new(LONG_TYPE, new IntValueNode(long.Parse(defaultValueFromConfig))),
+                STRING_TYPE => new(STRING_TYPE, new StringValueNode(defaultValueFromConfig)),
+                BOOLEAN_TYPE => new(BOOLEAN_TYPE, new BooleanValueNode(bool.Parse(defaultValueFromConfig))),
+                SINGLE_TYPE => new(SINGLE_TYPE, new SingleType().ParseValue(float.Parse(defaultValueFromConfig))),
+                FLOAT_TYPE => new(FLOAT_TYPE, new FloatValueNode(double.Parse(defaultValueFromConfig))),
+                DECIMAL_TYPE => new(DECIMAL_TYPE, new FloatValueNode(decimal.Parse(defaultValueFromConfig))),
+                DATETIME_TYPE => new(DATETIME_TYPE, new DateTimeType().ParseResult(DateTime.Parse(defaultValueFromConfig))),
+                BYTEARRAY_TYPE => new(BYTEARRAY_TYPE, new ByteArrayType().ParseValue(Convert.FromBase64String(defaultValueFromConfig))),
+                _ => throw new DataApiBuilderException(
+                    message: $"The parameter value {defaultValueFromConfig} provided in configuration cannot be converted to the type {paramValueType}",
```

Comment:
Should this exception be more like not supported parameter type?

Reply:
Sure, that's a more informative (and actionable) message.

Reply:
This might probably be a gap on our side - might need to add support for whatever data type that wasn't successfully parsed. Also would throwing a notsupported exception make more sense?